templates/esp-network-policy.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: esp-network-policy
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: esp
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  policyTypes:
+  - Ingress
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            app.kubernetes.io/name: gateway
+            app.kubernetes.io/instance: {{ .Release.Name }}
+      ports:
```

Comment:
I wonder if restricting the ingress/egress ports might be overcomplicating things? Is the "pod-to-pod" level not enough?

The Deployment / Pod already defines which ports should be exposed, and having to remember to match that up in a network policy might be a bit annoying (it's a shame you can't use the port's names here to at least avoid having to duplicate port numbers)

I'm also a bit unsure what will happen when traffic is being routed via the istio sidecar with the port-level restriction, since presumably the mTLS connections will pick their own ports or be 443 or something.

Reply:
ahh, maybe. I'm not wedded to restricting ports too—it just seemed trivial to add at the time.

Reply:
I think removing the ports might be a bad idea. Although it isn't used yet Dropwizard exposes metrics on a special port on the pod. We'd want the `metrics-system` pods to scrape this without also being able to talk to the normal app port.

I still don't know how it works with Istio though.

Reply:
the `mgmt` port is also used for healthchecks ... so might need to give api-server or kubelet or whatever it is that polls the liveness / readiness probe urls

Reply:
This doesn't seem to be a problem—the kubelet is the thing that performs the container livenss checks and it must have host-level access to the container.